tests/lax_control_flow_test.py

```diff
@@ -2928,6 +2928,17 @@ def body(carry, x):
     hlo_text = fn.lower(init).as_text('hlo')
     self.assertNotIn('4,1,2,2', hlo_text)
 
+  def test_cond_vmap_forwarding_doesnt_promote(self):
```

Comment:
Can we also test forwarding directly somehow? Maybe something along the lines of calling the cond and using `assertIs` to check input/output forwarding by object identity match?

Reply:
sounds good